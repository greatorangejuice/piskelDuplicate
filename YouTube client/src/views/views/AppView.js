export default class AppView {
  constructor(snippets) {
    this.snippets = snippets;
  }


  render() {
    const content = document.createElement('ul');
    content.innerHTML = this.titles.map(title => `<li>${title}</li>`).join('');
    document.body.appendChild(content);
  }

  func() {
    this.snippets.forEach(data => { // Приходит массив объектов. Проходим, деструктурируем.
      // записываем в DOM.
      let {title, decription, publishedAt} = data.snippet;
      const titleField = document.createElement('p');
      titleField.innerHTML = title;
      // И так со всеми элементами.
    });
  }
}
// Нужно запилить 15 блоков DOM, в каждый запихнуть всю нужную инфу, а уже потом пилить слайдер
// Осталось придумать, как это правильно мапами обработать.
// Придёт массив с объектами. А потом буду парсить.
