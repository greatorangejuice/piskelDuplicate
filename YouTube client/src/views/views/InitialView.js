import './initial-view.css';

export default class InitialView {
  static getInitialDom() {
    const appContainer = document.createElement('div');
    appContainer.className = 'application';
    document.body.appendChild(appContainer);

    const wrapper = document.createElement('div');
    wrapper.className = 'search-field-wrapper';

    const searchField = document.createElement('input');
    searchField.value = 'Nature';
    searchField.className = 'search-field';

    const startButton = document.createElement('button');
    startButton.innerText = 'Search';
    startButton.className = 'search-button';

    appContainer.appendChild(wrapper);
    wrapper.appendChild(searchField);
    wrapper.appendChild(startButton);

    const dataWrapper = document.createElement('div');
    dataWrapper.className = 'main-container';
    appContainer.appendChild(dataWrapper);

    const galleryContainer = document.createElement('div');
    galleryContainer.className = 'wrapper';
    dataWrapper.appendChild(galleryContainer);

    const swipeContainer = document.createElement('div');
    swipeContainer.className = 'gallery';
    galleryContainer.appendChild(swipeContainer);

    const sliderScriptLink = document.createElement('script');
    sliderScriptLink.type = 'text/javascript';
  }
}
