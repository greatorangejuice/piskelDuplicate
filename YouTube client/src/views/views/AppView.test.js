import AppView from './AppView';
import InitialView from './InitialView';

describe('AppView.prototype.render', () => {
  it('Should be an instance of function', () => {
    expect(AppView.prototype.render).toBeInstanceOf(Function);
  });

  it('Should be render correctly', () => {
    InitialView.getInitialDom();
    const context = [
      {
        id: { kind: 'youtube#video', videoId: '0I647GU3Jsc' },
        snippet: {
          publishedAt: '2018-08-24T19:00:01.000Z',
          channelId: 'UCpx_k19S2vUutWUUM9qmXEg',
          title: 'Imagine Dragons - Natural',
          description: 'Descr',
          thumbnails: {
            medium: {
              url: 'https://url.jpg',
            },
          },
        },
        views: '229481471',
      },
      {
        id: { kind: 'youtube#video', videoId: '0I647GU3Jsc' },
        snippet: {
          publishedAt: '2018-08-24T19:00:01.000Z',
          channelId: 'UCpx_k19S2vUutWUUM9qmXEg',
          title: 'Imagine Dragons - Natural',
          description: 'Descr',
          thumbnails: {
            medium: {
              url: 'https://url.jpg',
            },
          },
        },
        views: '229481471',
      },
    ];
    AppView.prototype.render(context);

    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
