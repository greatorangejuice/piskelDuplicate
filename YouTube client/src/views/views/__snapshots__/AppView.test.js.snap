// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppView.prototype.render Should be render correctly 1`] = `"<div class=\\"application\\"><div class=\\"search-field-wrapper\\"><input class=\\"search-field\\"><button class=\\"search-button\\"></button></div><div class=\\"main-container\\"><div class=\\"wrapper\\"><div class=\\"gallery\\"><div class=\\"youtube-item\\"><div class=\\"video-title\\"><a href=\\"https://www.youtube.com/watch?v=0I647GU3Jsc\\" target=\\"_blank\\"></a></div><img src=\\"https://url.jpg\\"><p></p><p></p><p></p><p class=\\"count-views\\"></p></div><div class=\\"youtube-item\\"><div class=\\"video-title\\"><a href=\\"https://www.youtube.com/watch?v=0I647GU3Jsc\\" target=\\"_blank\\"></a></div><img src=\\"https://url.jpg\\"><p></p><p></p><p></p><p class=\\"count-views\\"></p></div></div></div></div></div>"`;
